crates/flux-fhir-analysis/src/wf/sortck.rs

```diff
@@ -270,8 +256,58 @@ impl<'genv, 'tcx> InferCtxt<'genv, 'tcx> {
                     _ => Err(self.emit_field_not_found(&sort, *fld)),
                 }
             }
-            // (VR): Pretty sure we don't synthesize these?
-            _ => span_bug!(expr.span, "unexpected expr `{expr:?}` in synth expr"),
+            fhir::ExprKind::Constructor(Some(path), field_exprs, spread) => {
+                // if we have S then we can synthesize otherwise we fail
+                // first get the sort based on the path - for example S { ... } => S
+                // and we should expect sort to be a struct or enum app
+                let path_def_id = match path.res {
+                    ExprRes::Struct(def_id) | ExprRes::Enum(def_id) => def_id,
+                    _ => span_bug!(expr.span, "unexpected path in constructor"),
+                };
+                let sort_def = match self.genv.adt_sort_def_of(path_def_id) {
+                    Ok(sd) => sd,
+                    Err(_) => span_bug!(path.span, "Unexpected path in constructor"),
+                };
```

Comment:
This error needs to be propagated

```suggestion
                let sort_def = match self.genv.adt_sort_def_of(path_def_id).emit(&self.genv)?;
```